packages/ra-ui-materialui/src/form/SimpleFormConfigurable.tsx

```diff
@@ -31,27 +31,28 @@ export const SimpleFormConfigurable = ({
 
     React.useEffect(() => {
         // first render, or the preference have been cleared
-        const inputs = React.Children.map(props.children, (child, index) =>
-            React.isValidElement(child)
-                ? {
-                      index: String(index),
-                      source: child.props.source,
-                      label:
-                          child.props.source || child.props.label
-                              ? child.props.label
-                              : translate(
-                                    'ra.configurable.SimpleForm.unlabeled',
-                                    {
-                                        input: index,
-                                        _: `Unlabeled input #%{input}`,
-                                    }
-                                ),
-                  }
-                : null
-        ).filter(column => column != null);
+        const inputs =
+            React.Children.map(props.children, (child, index) =>
+                React.isValidElement(child)
+                    ? {
+                          index: String(index),
+                          source: child.props.source,
+                          label:
+                              child.props.source || child.props.label
+                                  ? child.props.label
+                                  : translate(
+                                        'ra.configurable.SimpleForm.unlabeled',
+                                        {
+                                            input: index,
+                                            _: `Unlabeled input #%{input}`,
+                                        }
+                                    ),
+                      }
+                    : null
+            )?.filter(column => column != null) ?? [];
```

Comment:
I think you should use an emptyArray const defined outside of the element to avoid rerenderings when the effect returns a different empty array object (modified 3 times)